internal/app/dashboards/model.go

```diff
@@ -0,0 +1,39 @@
+package dashboards
+
+import "platform-go-challenge/internal/app/assets"
+
+type Dashboard struct {
+	ID        uint32           `json:"id"`
+	UserID    uint32           `json:"-"`
+	Charts    StarredCharts    `json:"charts"`
+	Insights  StarredInsights  `json:"insights"`
+	Audiences StarredAudiences `json:"audiences"`
+}
+
+type StarredChart struct {
+	assets.Chart
+	Description string `json:"description"`
+}
+
+type StarredCharts []StarredChart
+
+type StarredInsight struct {
+	assets.Insight
+	Description string `json:"description"`
+}
+
+type StarredInsights []StarredInsight
+
+type StarredAudience struct {
```

Comment:
Do you see any issues using a model for both DB and HTTP interfaces? How you could separate those two entities if it was necessary? What would change on both HTTP and Repository ends?

Reply:
Yes. Since we may have some struct fields that we don't want to return or the actual structure of the model may be different in many ways we should have different models sometimes even per call or per end user. 
I would most probably create view models with custom marshal methods and constructors that used the DB models as parameters. The end result would have any kind of structure or fields we needed without having to use different implementations each time.